docs/principles-for-package-repository-security.md

```diff
@@ -0,0 +1,106 @@
+  - The package repository supports providing [build provenance](https://repos.openssf.org/build-provenance-for-all-package-registries) for packages.
+
+### General Capabilities
+
+Applies to: all package repositories
+
+- Level 1
+  - The package repository has published a vulnerability disclosure policy allowing security researchers to report vulnerabilities affecting the package repository, and receive legal safe harbor when doing so. The package repository should strive to remediate reported vulnerabilities within 90 days.
+  - The package repository takes steps to prevent typosquatting attacks around package names. This may include namespacing, detecting similar names, or other actions.
+- Level 2
+  - The package repository has an unpublish policy to prevent specific versions of a package from being replaced, and, if a package is deleted, prevent others from reusing the same package name and version.
+  - The package repository allows users to report a package as suspicious or malicious via the registry UI, tooling CLI, and via API.
+  - The package repository takes steps to detect malware, including scanning for known malware code snippets and hashes.
+- Level 3
+  - Undergo periodic cloud security review (and optionally publish remediations once they are complete)
+  - The package repository publishes an event transparency log to enable detection of anomalous behaviors (e.g. [replicate.npmjs.com](https://replicate.npmjs.com/)).
+  - Advisories for malicious packages are published using the OSV schema and aggregated by the OSV service (e.g. [here](https://github.com/ossf/malicious-packages))
+
+### CLI Tooling
+
+Applies to: all package management ecosystems, although here we are primarily focused on the CLI tooling instead of the registry itself.
+
+- Level 1
+  - The CLI allows installing dependencies that are pinned based on hash, version, etc.
+- Level 2
+  - The package repository warns of known security vulnerabilities in dependencies, in both the package repository UI and CLI tools when installing.
+- Level 3
+  - The CLI has functionality to produce SBOMs.
+  - The package repository is able to reduce false positives of identified vulnerabilities using static analysis to understand whether a vulnerable code path is actually reachable. (e.g. [Golang's vulncheck feature](https://go.dev/security/vuln))
+
+## Resources / Inspiration
+
+- <https://github.com/psf/fundable-packaging-improvements/blob/master/FUNDABLES.md>
+- <https://github.com/ossf/wg-securing-software-repos/blob/main/survey/2022/results.csv>
+- <https://docs.npmjs.com/threats-and-mitigations>
+- <https://www.microsoft.com/en-us/research/uploads/prod/2018/03/build-systems.pdf> (as a framework for comparing options in a problem space)
```

Comment:
```suggestion
- <https://www.microsoft.com/en-us/research/uploads/prod/2018/03/build-systems.pdf> (as a framework for comparing options in a problem space)

Note: CISA does not endorse any commercial entity, product, company, or service, including any entities, products, or services linked within this document. Any reference to specific commercial entities, products, processes, or services by service mark, trademark, manufacturer, or otherwise, does not constitute or imply endorsement, recommendation, or favoring by CISA.
```